ClinGen/clincoded
# Connects to #1104. Adding adjectives to mode of inheritance.

**Technical notes:**
1. This PR is primarily for the changes made to the mode of inheritance, in which adjectives are added to the following core MOIs: `Autosomal dominant inheritance (HP:0000006)`, `Autosomal recessive inheritance (HP:0000007)`, `Mitochondrial inheritance (HP:0001427)`, `X-linked inheritance (HP:0001417)` and `Other`.
2. The changes made to the mode of inheritance also include moving the following core MOIs to be adjectives under the `Other` core MOI: `Y-linked inheritance (HP:0001450)`, `Somatic mutation (HP:0001428)` and `Multifactorial inheritance (HP:0001426)`.
3. The mode of inheritance changes will affect the GDM creation in GCI as well as the MOI association with interpretation in VCI.
4. Curators are not required to select an adjective for any given core MOIs (although they could).
5. HPO terms in any given adjectives have been suppressed from the display only.
6. `modeInheritance` and `modeInheritanceAdjective` are used to store the core MOI and adjective values separately.

**Steps to test:**
1. Login and proceed to **Create New Gene Disease Record** page (e.g. ^/create-gene-disease/).
2. After entering the required **HGNC gene symbol** and **Orphanet ID**, select any core MOIs from the **Mode of Inheritance** dropdown options (except "Unknown") and expect to see the **adjectives** dropdown menu to be enabled with their respective lists of options.
3. When an adjective is selected for a given core MOI, save the GDM and expect to see the adjective being appended to the core MOI in parentheses as the subtitle on the subsequent record page (e.g. Curation-Central).
4. Now go to the VCI's variation selection interface and use 12345 as the ClinVar ID.
5. Start a new interpretation on the subsequent VCI's evidence view page, and then select some criteria modifiers so that pathogenicity can be calculated.
6. Click on the **Inheritance+** button and bring up the modal. Similarly, select any core MOIs from the **Mode of Inheritance** dropdown options (except "Unknown") and expect to see the **adjectives** dropdown menu to be enabled with their respective lists of options.
7. When an adjective is selected for a given core MOI, save the modal and expect to see the adjective being appended to the core MOI in parentheses in the subtitle on the VCI page.
8. Click on the **View Summary** button and proceed to the VCI interpretation summary page. Expect to see the adjective being appended to the core MOI in parentheses in the provision panel. If the core MOI has an HPO term, it is displayed as an external linkout. Otherwise, it is displayed as plain text.
9. Click on the **Return to Interpretation** button to go back to the VCI interpretation interface. Bring up the mode of inheritance modal again. Expect to see the previously selected core MOI and its adjective being shown in the modal dropdown menus. Also expect to be able to change to other core MOIs and/or adjectives in the modal.
